core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -283,17 +291,29 @@ public static String subtractPaths(String path, String prefix) throws InvalidPat
    * @throws InvalidPathException when the path or prefix is invalid
    */
   public static boolean hasPrefix(String path, String prefix) throws InvalidPathException {
-    String[] pathComponents = getPathComponents(path);
-    String[] prefixComponents = getPathComponents(prefix);
-    if (pathComponents.length < prefixComponents.length) {
-      return false;
+    if (prefix.isEmpty() || path.isEmpty()) {
+      throw new InvalidPathException(ExceptionMessage.PATH_INVALID.getMessage(""));
     }
-    for (int i = 0; i < prefixComponents.length; i++) {
-      if (!pathComponents[i].equals(prefixComponents[i])) {
-        return false;
-      }
+
+    if (prefix.equals("/")) {
+      return true;
+    }
+
+    if (prefix.endsWith("/") && !path.endsWith("/")) {
+      path = normalizePath(path, "/");
+    }
+    // Firstly, we check if the `path` starts with `prefix`,
+    if (path.startsWith(prefix)) {
+      // Then, check if path and prefix has the same length;
+      boolean isPathLengthEqualsPrefix = path.length() == prefix.length();
+      // This condition is to include the case like `prefix: /a/b/, path: /a/b/c/`
+      boolean isPathPrefixEnd = prefix.endsWith("/") && path.charAt(prefix.length() - 1) == '/';
+      // This condition is to exclude the case like `prefix: '/a/b/c', path: '/a/b/ccc'`
+      boolean isPathValidAfterThePrefix =
+          path.length() > prefix.length() && path.charAt(prefix.length()) == '/';
+      return isPathLengthEqualsPrefix || isPathPrefixEnd || isPathValidAfterThePrefix;
```

Comment:
```suggestion
    // If prefix is "/a/" and path is "/a", make path "/a/"
    if (prefix.endsWith("/") && !path.endsWith("/")) {
      path = normalizePath(path, "/");
    }
    if (path.startsWith(prefix)) {
      return path.length() == prefix.length()  // path == prefix
          // Include cases like `prefix=/a/b/, path=/a/b/c/`
          || (prefix.endsWith("/") && path.charAt(prefix.length() - 1) == '/')
          // Exclude cases like `prefix=/a/b/c, path=/a/b/ccc`
          || (path.charAt(prefix.length()) == '/');
```

Reply:
Actually can you also check the length of two strings?
```
if (prefix.endsWith("/") && !path.endsWith("/") && (path.length() == prefix.length() + 1)) {
```

Reply:
The last `path.length() > prefix.length()` is not needed because it's implicitly checked already